### PR TITLE
Video-48-Edit-User

### PR DESCRIPTION
**Pull request by basir**

### CODE REVIEW

File: backend/routers/userRouter.js [hunk 1]
```diff
@@ -118,4 +118,23 @@ userRouter.delete(
...
+userRouter.put(
+  '/:id',
+  isAuth,
+  isAdmin,
+  expressAsyncHandler(async (req, res) => {
+    const user = await User.findById(req.params.id);
+    if (user) {
+      user.name = req.body.name || user.name;
+      user.email = req.body.email || user.email;
+      user.isSeller = req.body.isSeller || user.isSeller;
```
Comment (payanazo) [inline backend/routers/userRouter.js:130]: there should be another type of evaluation. req.body.name can return a false value causing that the router will always save the previous state value. Remove the or condition and you will be able to change IsAdmin and IsSeller value from true to false.

Comment (sai-potala) [reply backend/routers/userRouter.js:130]: use this logic

user.isSeller =
        req.body.isSeller === user.isSeller ? user.isSeller : req.body.isSeller;
      
      user.isAdmin =
        req.body.isAdmin === user.isAdmin ? user.isAdmin : req.body.isAdmin;

Comment (DiegoRojasVera) [reply backend/routers/userRouter.js:130]: How would it have to be for it to be edited?

Comment (SylvOne) [reply backend/routers/userRouter.js:130]: **DiegoRosasVera** the response is given by **saivenkatabhaskarateja-potala-au9** : 

In the file "userRouter.js" at the end in "userRouter.put" in the condition 
```
if (user) {
      user.name = req.body.name || user.name;
      user.email = req.body.email || user.email;
      user.isSeller = req.body.isSeller || user.isSeller;
user.isAdmin = req.body.isAdmin || user.isAdmin;
      const updatedUser = await user.save();
      res.send({ message: 'User Updated', user: updatedUser });
    }
```
change these two lines 
```
user.isSeller = req.body.isSeller || user.isSeller; 
user.isAdmin = req.body.isAdmin || user.isAdmin;
``` 
to : 
```
user.isSeller = req.body.isSeller === user.isSeller ? user.isSeller : req.body.isSeller;
user.isAdmin = req.body.isAdmin === user.isAdmin ? user.isAdmin : req.body.isAdmin;
```

and it's ok. Thanks to **saivenkatabhaskarateja-potala-au9**

Comment (marothi-codes) [reply backend/routers/userRouter.js:130]: You can also do the same for the username.
`user.name = req.body.name === user.name ? user.name : req.body.name;`

This way one can sleep like a baby knowing that the code is bound to work...

**Pull request merged**